src/lib.rs

```diff
@@ -149,15 +149,89 @@ mod memlock {
 /// Type alias for a vector that stores just bytes
 pub type SecStr = SecVec<u8>;
 
+/// Wrapper for a vector that stores a valid UTF-8 string
+#[derive(Clone, Eq)]
+pub struct SecUtf8(SecVec<u8>);
+
+impl SecUtf8 {
+    pub fn unsecure(&self) -> &str {
+        unsafe { std::str::from_utf8_unchecked(self.0.unsecure())}
+    }
+}
+
+impl PartialEq for SecUtf8 {
+    fn eq(&self, other: &SecUtf8) -> bool {
+        // use implementation of SecVec
+        self.0 == other.0
+    }
+}
+
+impl fmt::Debug for SecUtf8 {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.write_str("***SECRET***").map_err(|_| { fmt::Error })
+    }
+}
+
+impl fmt::Display for SecUtf8 {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.write_str("***SECRET***").map_err(|_| { fmt::Error })
+    }
+}
+
+impl<U> From<U> for SecUtf8 where U: Into<Vec<u8>> {
+    fn from(s: U) -> SecUtf8 {
+        SecUtf8(SecVec::new(s.into()))
+    }
+}
+
+impl From<SecUtf8> for String {
+    fn from(mut s: SecUtf8) -> String {
+        let content = std::mem::replace(&mut s.0.content, Vec::new());
+        unsafe { String::from_utf8_unchecked(content) }
+    }
+}
+
+#[cfg(feature = "serde")]
+impl serde::Serialize for SecUtf8 {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(self.unsecure())
+    }
+}
+
+#[cfg(feature = "serde")]
+impl<'de> serde::Deserialize<'de> for SecUtf8 {
+    fn deserialize<D>(deserializer: D) -> Result<SecUtf8, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        struct SecUtf8Visitor;
+        impl<'de> serde::de::Visitor<'de> for SecUtf8Visitor {
+            type Value = SecUtf8;
+            fn expecting(&self, formatter: &mut ::std::fmt::Formatter) -> ::std::fmt::Result {
+                write!(formatter, "an utf-8 encoded string")
+            }
+            fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                Ok(SecUtf8::from(v.to_string()))
+            }
+        }
+        deserializer.deserialize_string(SecUtf8Visitor)
+    }
+}
 
-/// A data type suitable for storing sensitive information such as passwords and private keys in memory, that implements:  
-/// 
-/// - Automatic zeroing in `Drop`  
```

Comment:
The trailing whitespace here — in doc comments specifically — was intentional: two spaces at the end signify a line break (`<br>`). Though this is a bulleted list anyway, I'm not sure this is required…

Reply:
I'm not sure, this is how it looks like
![screenshot 2019-01-14 at 12 54 18](https://user-images.githubusercontent.com/1214337/51111203-94ab5f80-17fb-11e9-9d06-dadc2f29a062.png)